### DISCUSSION TITLE
Negative frequencies in SCPH calculations

### DISCUSSION THREAD

**Discussion by RicardoCecconello** [ttadano/alamode#217] [Q&A]
Dear Dr. Tadano,

I'm finding negative frequencies for converged SCPH calculations, even though they are not present in the harmonic calculations.
This is the result of the harmonic calculation (fitting error = 0.6 %), with displacement patterns:

![PBE_bands_harm](https://github.com/user-attachments/assets/71655b6e-855c-461d-8d22-4e64367b1f5c)

This is the result of the SCPH calculation at 0 K, with cutoff None 15 8, NBODY 232, fixing the 2nd order IFCs with the result above, and the DFSET was made with 320 structures randomly displaced:

![PBE_bands_scph](https://github.com/user-attachments/assets/c440169e-4231-4bd6-a3a6-6734cbc18e46)

This result is similar to temperatures up to 50 K, and the negative frequencies also appear for cutoff None 15 7 6 6 and NBODY 23322. The fitting error (with ENET) is about 4 %. Changing the SCPH meshes or MIXALPHA doesn't make it better, and with SELF_OFFDIAG = 1 it doesn't converge. Any ideas or suggestions on the reason for that?

**Comment by ttadano** [comment]
Please tell us 

- the supercell sizes for IFC2 and IFC4
- `KMESH_INTERPOLATE` and `KMESH_SCPH`.

**Reply by RicardoCecconello** [reply]
The supercell is 4x4x2 for all IFCs, and the meshes are 4 4 2.

**Reply by ttadano** [reply]
Thanks. The `KMESH_*` values look reasonable. 

In addition to the negative frequency along the Z-l1 line, small imaginary pockets appear near the \Gamma point in the SCPH band structure. This indicates the accuracy of the anharmonic IFC is not enough, and you may need to increase the cutoff radius for the forth-order term to resolve it.

In some cases, similar problem could be resolved by using a coarse `KMESH_*` values, say `KMESH_INTERPOLATE = 2 2 2` or `2 2 1`.

Also, did you use the same `NONANALYTIC` option for the harmonic and SCPH calculations? If not, please do so.

**Reply by RicardoCecconello** [reply]
I used the default value of `NONANALYTIC` for both of them.

Using 12 12 6 for `KMESH_INTERPOLATE` and `KMESH_SCPH` gives the following:


![12126](https://github.com/user-attachments/assets/086bf15d-238c-4c53-85c8-92b5fb11f57f)

Those negative frequencies "move" to the Z-G direction, and those imaginary pockets near G get bigger. For `KMESH_INTERPOLATE = 2 2 1` and `KMESH_SCPH = 4 4 2` the spectrum is

![221](https://github.com/user-attachments/assets/c5ad3fc1-7ce0-42e5-bc83-3f915d93d66f)

and similarly for `KMESH_INTERPOLATE = 2 2 2` and `KMESH_SCPH = 2 2 2`. 

We'll try increasing the 4th order cutoff.

**Comment by liudadie12138** [comment]
Hello, have you resolved the issue by increasing the cutoff? I have encountered the same problem recently.

**Reply by RicardoCecconello** [reply]
Hello! We changed a lot of thing since then, and now we use NBODY 23322 and cutoff None 15 8 6 6, wich gives good results. One of the thing is the size of the DFSET; now we use 600 structures.
